stl/inc/ranges

```diff
@@ -4748,6 +4748,566 @@ namespace ranges {
+                requires sized_sentinel_for<_Base_iter, _Base_iter> {
+                return (_Left._Current - _Right._Current + _Left._Missing - _Right._Missing) / _Left._Count;
+            }
+
+            _NODISCARD_FRIEND constexpr difference_type operator-(default_sentinel_t, const _Iterator& _Right) noexcept(
+                noexcept(_Right._End - _Right._Current)) /* strengthened */
+                requires sized_sentinel_for<_Base_sen, _Base_iter> {
+                return _Div_ceil(_Right._End - _Right._Current, _Right._Count);
+            }
+
+            _NODISCARD_FRIEND constexpr difference_type operator-(const _Iterator& _Left, default_sentinel_t) noexcept(
+                noexcept(_Left._End - _Left._Current)) /* strengthened */
+                requires sized_sentinel_for<_Base_sen, _Base_iter> {
+                return -_Div_ceil(_Left._End - _Left._Current, _Left._Count);
+            }
+        };
+
+    public:
+        // clang-format off
+        chunk_view() requires default_initializable<_Vw> = default;
+        // clang-format on
+
+        constexpr explicit chunk_view(_Vw _Range_, const range_difference_t<_Vw> _Count_) noexcept(
+            is_nothrow_move_constructible_v<_Vw>) /* strengthened */
+            : _Range(_STD move(_Range_)), _Count{_Count_} {
+#if _CONTAINER_DEBUG_LEVEL > 0
+            _STL_VERIFY(_Count > 0, "n must be greater than 0");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        }
+
+        _NODISCARD constexpr _Vw base() const& noexcept(is_nothrow_copy_constructible_v<_Vw>) /* strengthened */
+            requires copy_constructible<_Vw> {
+            return _Range;
+        }
+
+        _NODISCARD constexpr _Vw base() && noexcept(is_nothrow_move_constructible_v<_Vw>) /* strengthened */ {
+            return _STD move(_Range);
+        }
+
+        // clang-format off
+        _NODISCARD constexpr auto begin() noexcept(
+            noexcept(_RANGES begin(_Range)) && is_nothrow_move_constructible_v<iterator_t<_Vw>>) /* strengthened */
+            requires (!_Simple_view<_Vw>) {
+            // clang-format on
+            return _Iterator<false>{this, _RANGES begin(_Range)};
+        }
+
+        _NODISCARD constexpr auto begin() const
+            noexcept(noexcept(_RANGES begin(_Range))
+                     && is_nothrow_move_constructible_v<iterator_t<const _Vw>>) /* strengthened */
+            requires forward_range<const _Vw> {
+            return _Iterator<true>{this, _RANGES begin(_Range)};
+        }
+
+        // clang-format off
+        _NODISCARD constexpr auto end() noexcept(noexcept(_RANGES distance(_Range))
+            && noexcept(_RANGES begin(_Range)) && is_nothrow_move_constructible_v<iterator_t<_Vw>>) /* strengthened */
+            requires (!_Simple_view<_Vw>) {
+            // clang-format on
+            if constexpr (common_range<_Vw> && sized_range<_Vw>) {
+                const auto _Missing = (_Count - _RANGES distance(_Range) % _Count) % _Count;

```

Comment:
This is the correct implementation according to the standard, but it's a bad implementation since it requires two divrems. It should probably be a piecewise function:

```suggestion
                auto _Missing = _Count - _RANGES distance(_Range) % _Count;
                if (_Missing == _Count) _Missing = 0;
```

proof of equivalence is left as an exercise to the reader (I did it on a whiteboard and if you want a picture I can send it to you on discord)

Reply:
Yeah that is obviously correct. I have adopted it to
```cpp
                const auto _Missing = _Count - _RANGES distance(_Range) % _Count;
                if (_Missing == _Count) {
                    return _Iterator<false>{this, _RANGES end(_Range)};
                }
                return _Iterator<false>{this, _RANGES end(_Range), _Missing};
```

Reply:
\[insert complaining that there isn't a [`euclidean_remainder`] function in the standard\]

[`euclidean_remainder`]: https://doc.rust-lang.org/stable/std/primitive.i32.html#method.rem_euclid